### PR TITLE
etcd-signer-server: allow server to serve TLS based on SNI

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Currently the etcd-signer-server does not support serving TLS traffic based on SNI. Therefore, since openshift's kube-apiserver serves
traffic on api.$cluster_domain and api-int.$cluster_domain [1] and  because during bootstrapping when etcd-signer-server is mimic-ing the kube-apiserver to
sign the etcd clients certificates it can only serve traffic on single domain, external clients trying to connect to `:6443` from api.$cluster_domain see errors like,

```console
time="2019-04-24T13:25:11-07:00" level=debug msg="Still waiting for the Kubernetes API: Get https://api.adahiya-0.tt.testing:6443/version?timeout=32s: x509: certificate is valid for api-int.adahiya-0.tt.testing, not api.adahiya-0.tt.testing"
```
as the etcd-signer-server is using certs for `api-int.$cluster_domain` as internal clients ie etcd agent is contacting it on that domain, the external clients ie the installer hits the etcd-signer on `api.$cluster_domain`

Allowing etcd-signer-server to accept multiple certs and serve TLS based on SNI allows it to correctly mimic the kube-apiserver's capability.

[1]: https://github.com/openshift/installer/pull/1633

### CODE REVIEW

File: cmd/kube-etcd-signer-server/serve.go [hunk 1]
```diff
@@ -62,8 +62,10 @@ func validateServeOpts(cmd *cobra.Command, args []string) error {
 		return errors.New("no signer CA flags passed one cert/key pair is required")
 	}
 
-	if serveOpts.sCrtFile == "" || serveOpts.sKeyFile == "" {
-		return errors.New("both --servcrt and --servkey are required flags")
+	if cl, kl := len(serveOpts.sCrtFiles), len(serveOpts.sKeyFiles); cl == 0 || kl == 0 {
+		return errors.New("at least one pair of --servcrt and --servkey is required")
+	} else if cl != kl {
+		return errors.New("mismatch pairs of --servcrt and --servkey")
```
Comment (wking) [inline cmd/kube-etcd-signer-server/serve.go]: This might possibly be read as "key A is not a valid key for cert A".  Maybe rephrase to:

```go
return errors.Errorf("%d --servercrt does not match %d --servkey", cl, kl)
```

or something?

File: pkg/certsigner/signer.go [hunk 2]
```diff
@@ -77,14 +78,17 @@ type CertSigner struct {
...
 // Config holds the configuration values required to start a new signer
 type Config struct {
 	// SignerCAFiles
 	SignerCAFiles
-	// ServerCertFile is the file location of the server certificate
-	ServerCertFile string
-	// ServerKeyFile is the file location of the server private key
-	ServerKeyFile string
+	// ServerCertKeys is a list of server certificates for serving on TLS based on SNI
+	ServerCertKeys []CertKey
```
Comment (wking) [inline pkg/certsigner/signer.go:91]: nit: Using `[]tls.Certificate` here (and loading the cert/key pairs in `cmd/kube-etcd-signer-server/serve.go`) would let us simplify the `pkg/certsigner` API slightly by leaning on stdlib types instead of defining our own.  But whatever, folks are unlikely to be using `pkg/certsigner` outside of this repository ;).

**Comment by wking** [comment]
How do we get this picked up in openshift/kubecsr?

**Comment by hexfusion** [comment]
@wking I will take care of it in a few

**Comment by hexfusion** [comment]
https://github.com/openshift/kubecsr/pull/14
